elytron/src/main/java/org/wildfly/extension/elytron/KeyPairUtil.java

```diff
@@ -0,0 +1,136 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.InvalidParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+
+import static org.wildfly.extension.elytron._private.ElytronSubsystemMessages.ROOT_LOGGER;
+
+public class KeyPairUtil {
+
+    public static final String RSA_ALGORITHM = "RSA";
+    public static final String DSA_ALGORITHM = "DSA";
+    public static final String EC_ALGORITHM = "EC";
+
+    public static KeyPair generateKeyPair(String algorithm, int size) throws NoSuchAlgorithmException {
+        KeyPairGenerator keyPairGenerator;
+
+        switch (algorithm) {
+            case RSA_ALGORITHM: {
+                /* Size must range from 512 to 16384. Default size: 2048
+                 * see: https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2
+                 */
+                size = (512 <= size && size <= 16384) ? size : 2048;
+                break;
+            }
+            case DSA_ALGORITHM: {
+                /* Size must be multiple of 64 ranging from 512 to 1024, plus 2048 and 3072. Default size: 2048
+                 * see: https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-3A80CC46-91E1-4E47-AC51-CB7B782CEA7D
+                 */
+                size = (512 <= size && size <= 1024 && (size % 64) == 0) || size == 2048  || size == 3072 ? size : 2048;
+                break;
+            }
+            case EC_ALGORITHM: {
+                /* Size must range from 112 to 571. Default size: 256
+                 * see: https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-091BF58C-82AB-4C9C-850F-1660824D5254
+                 */
+                size = (112 <= size && size <= 571) ? size : 256;
+                break;
+            }
+            default: {
+                algorithm = RSA_ALGORITHM;
+                size = 2048;
+                break;
+            }
+        }
+
+        try {
+            keyPairGenerator = KeyPairGenerator.getInstance(algorithm);
+        } catch (NoSuchAlgorithmException e) {
+            throw ROOT_LOGGER.unknownKeyPairAlgorithm(algorithm);
+        }
+        try {
+            keyPairGenerator.initialize(size, new SecureRandom());
+        } catch (InvalidParameterException e) {
+            throw ROOT_LOGGER.invalidKeyPairKeySize(e.getMessage());
+        }
+        return keyPairGenerator.generateKeyPair();
+    }
+
+    public static KeyPair parseKeyPair(String privateKeyContent, String publicKeyContent, ElytronFilePasswordProvider passwordProvider) {
+        KeyPair keyPair;
+        try {
+            keyPair = Pem.parsePemOpenSSHContent(CodePointIterator.ofString(privateKeyContent), passwordProvider).next().tryCast(KeyPair.class);
```

Comment:
We might want to consider having a different parameter for specifying OpenSSH format so we don't try to guess the format. As an example, take a look at how this was done here:

https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/client/src/main/resources/schema/elytron-client-1_7.xsd